internal/datastores/vsan_datastore_subresource.go

```diff
@@ -0,0 +1,70 @@
+/* Copyright 2023 VMware, Inc.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package datastores
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	validation_utils "github.com/vmware/terraform-provider-vcf/internal/validation"
+	"github.com/vmware/vcf-sdk-go/models"
+)
+
+// VsanDatastoreSchema this helper function extracts the VSAN Datastore schema, so that
+// it's made available for both Domain and Cluster creation.
+func VsanDatastoreSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"datastore_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Datastore name used for cluster creation",
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"license_key": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "License key for the vSAN data store to be applied in vCenter",
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"failures_to_tolerate": {
+				Type:         schema.TypeInt,
+				Required:     true,
```

Comment:
Raj, I'm developing the provider against VCF 4.5.0, so that we can have support for that version. 5.0 wasn't out when I started. In the 4.5.0 API it shows the parameter as required.

Reply:
Either way I'm making it Optional and will let the validation API fail.